sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_message_backcompat.py

```diff
@@ -63,11 +65,13 @@ def __init__(self, message, **kwargs):
         )
         self.application_properties = (
             self._message.application_properties
-            if any(self._message.application_properties)
+            if self._message.application_properties and any(self._message.application_properties)
```

Comment:
This sort've implies that there's been a change where we have from "empty application properties" to "None".
Has something actually changed here? Or was the logic always broken? Wouldn't the latter be unlikely given that not all our tests have app properties or annotations do they?

Reply:
Sorry, this one's on me🤦This logic was broken here before for the LegacyMessage in EH.

This was added during the uamqp switch changes. We updated the outgoing pyamqp/uamqp.Message creation method, so that application_properties and annotations would be set to None if those attributes weren't populated in the AmqpAnnotatedMessage. So, these could technically be None and this would've resulted in an error if the user called EventData.message on an already sent EventData that didn't have any application_properties/annotations. I didn't add a test for calling EventData.message on an already sent EventData, but we did in ServiceBus.